src/main/java/com/cegeka/academy/service/challenge/ChallengeServiceImp.java

```diff
@@ -150,6 +153,29 @@ public ChallengeDTO getChallengeById(long id) throws NotFoundException {
         return ChallengeMapper.convertChallengeToChallengeDTO(challenge);
     }
 
+    @Override
+    public List<UserChallengeDTO> getChallengeRanking(Long challengeId, String sortingParam) throws NotFoundException {
+        List<UserChallenge> userChallengeList = userChallengeRepository.findAllByChallengeId(challengeId);
+
+        if(userChallengeList.isEmpty())
+        {
+            throw new NotFoundException().setMessage("Nu exista participanti la challenge-ul: " + challengeId);
+        }
+
+        if(sortingParam.equals("name"))
+        {
+            userChallengeList.sort((o1, o2) -> o1.getUser().getLastName().compareToIgnoreCase(o2.getUser().getLastName()));
+        }
+
+        if(sortingParam.equals("points"))
+        {
+            userChallengeList.sort((o1, o2) -> -Double.compare(o1.getPoints(), o2.getPoints()));
+        }
+
```

Comment:
Iti ofer doua solutii:
1. Scoate o metoda separata partea cu sortarea. O poti denumi sortUserChallengeByCriteria, iar "name", "points" ar trebui sa fie un enum.
2. Fa un enum cu "name", "points" iar in acel enum defineste o metoda abstract in interior acelui enum care sa se numeasca sort. Cand intr-un enum faci o metoda abstract, o sa te oblige sa o implementezi, iar acolo o sa pui logica de sortare in functie de fiecare tip. 

Te rog sa citesti pe net despre metode in enum. Aici gasesti un exemplu:

https://dzone.com/articles/java-enums-how-to-make-much-more-useful